### DISCUSSION TITLE
xstate machines with vue apollo composable

### DISCUSSION THREAD

**Discussion by sschneider-ihre-pvs** [statelyai/xstate#3747] [Q&A]
I am a bit at a loss. I set up a basic machine, added `provideApolloClient` to make `useQuery` usable outside of setup. I then try to invoke a `useQuery` but actually what `useQuery` returns is an object `{ result, onResult, error, .....}` where the result will have the result when the request is done and error if there was an error and so on.

How do I handle that in the state machine?

**Comment by davidkpiano** [comment]
Can you share some of the current code so we can work from it?
